### ISSUE TITLE
It seems streaming requests are buffered in the proxy container

### ISSUE THREAD

**Issue by mattiasnordqvist** [villadora/express-http-proxy#236]
We've used the express-http-proxy up until now, but it seems it doesn't handle streaming very well, does it?

When we download big files through the proxy, the node server is starting to use up a lot of memory, and we don't get the http response until the whole file is written to memory in node.

I think this code in sendProxyRequest.js is what is causing the memory build up.

```
  return new Promise(function(resolve, reject) {
    var protocol = Container.proxy.requestModule;
    var proxyReq = protocol.request(reqOpt, function(rsp) {
      var chunks = [];
      rsp.on('data', function(chunk) { chunks.push(chunk); });
      rsp.on('end', function() {
        Container.proxy.res = rsp;
        Container.proxy.resData = Buffer.concat(chunks, chunkLength(chunks));
        resolve(Container);
      });
      rsp.on('error', reject);
    });
```

The promise is not resolved until we reach the end of the response stream. :(

**Comment by monkpow** [comment]
Yep, you're correct.    I recently reorganized the library into workflow steps, and my hope is that will allow us to swap this out for a streaming module when the file size is large or you have no interest in decorating the response.   But the current state is as you describe it.

**Comment by littletwolee** [comment]
hi @monkpow ,we have a problem like @mattiasnordqvist, so can you tell us, how much your cache limit of large file

**Comment by mattiasnordqvist** [comment]
It doesn't look like there is a specified limit, really. @littletwolee

**Comment by littletwolee** [comment]
@mattiasnordqvist 
I hope so,it look like have a cache.So, if we upload or download a big file, it need to cache by itself.I think this is why slowly.
